### ISSUE TITLE
Add support for disabled cron expression "-"

### ISSUE THREAD

**Issue by donalmurtagh** [kagkarlsson/db-scheduler#216]
According to the docs, db-scheduler supports

> Spring-style cron-expression.

However, a Spring cron expression of "-" means [the task will never run](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/scheduling/annotation/Scheduled.html#cron--), which is useful for disabling a task in a local/test environment.

I tried this expression with db-scheduler, but it caused this exception:

```
Caused by: java.lang.IllegalArgumentException: Cron expression contains 1 parts but we expect one of [6]
	at com.github.kagkarlsson.shaded.cronutils.parser.CronParser.parse(CronParser.java:120) ~[db-scheduler-10.3.jar:na]
	at com.github.kagkarlsson.scheduler.task.schedule.CronSchedule.<init>(CronSchedule.java:46) ~[db-scheduler-10.3.jar:na]
	at com.github.kagkarlsson.scheduler.task.schedule.CronSchedule.<init>(CronSchedule.java:56) ~[db-scheduler-10.3.jar:na]
	at com.github.kagkarlsson.scheduler.task.schedule.Schedules.cron(Schedules.java:39) ~[db-scheduler-10.3.jar:na]
```

**Comment by kagkarlsson** [comment]
Interesting. Yeah, that is not supported. I think I primarily meant that seconds are supported, so I probably should update the doc.

Not sure if it is possible to support that scenario just yet. When we add support for pausing/stopping an execution in db-scheduler, this would be a nice addition

**Comment by s-seidel** [comment]
We currently use this workaround, which is okay but not perfect:
```
    @Bean
    Task<Void> triggerCleanUp(final CleanUpService service,
            @Value("${cleanup.cron.expression}") final String cronPattern) {
        if (cronPattern == null || cronPattern.equals("-")) {
            return Tasks.custom("cleanup", Void.class).execute((i, c) -> new OnCompleteRemove<>());
        }
        return Tasks.recurring("cleanup", Schedules.cron(cronPattern)).execute((instance, ctx) -> {
            service.cleanUp();
        });
    }
```

**Comment by donalmurtagh** [comment]
This is the workaround that I use. Each `taskInvitationsSender` and `autocompleteEventsUpdater` beans are jobs that are run by the scheduler.

```java
@Configuration
public class ClusteredSchedulerConfiguration {

    private static final String DISABLED_CRON_EXPRESSION = "-";

    private static final Logger logger = LoggerFactory.getLogger(ClusteredSchedulerConfiguration.class);

    /**
     * There's no obvious way to create a schedule that will never execute (e.g. if you create a cron expression with a
     * date in the past, an exception is thrown), but in practical terms a task that uses this schedule will never be
     * executed.
     */
    private static final Schedule NULL_SCHEDULE = Schedules.fixedDelay(Duration.ofDays(Short.MAX_VALUE));

    @Bean
    Task<Void> scheduleTaskInvitationsSender(ScheduledTask taskInvitationsSender,
                                             @Value("${cron.task-invitations-sender}") String schedule) {
        return scheduleTask(schedule, taskInvitationsSender);
    }

    @Bean
    Task<Void> scheduleAutocompleteEvents(ScheduledTask autocompleteEventsUpdater,
                                         @Value("${cron.autocomplete-events}") String schedule) {
        return scheduleTask(schedule, autocompleteEventsUpdater);
    }

    private Task<Void> scheduleTask(String cronExpression, ScheduledTask task) {
        boolean isTaskDisabled = DISABLED_CRON_EXPRESSION.equals(cronExpression);
        Schedule schedule;

        if (isTaskDisabled) {
            // A workaround for the fact that db-scheduler doesn't support Spring's disabled cron expression
            // https://github.com/kagkarlsson/db-scheduler/issues/216
            logger.warn("The {} scheduled task is disabled in the current environment", task.getTaskName());
            schedule = NULL_SCHEDULE;
        } else {
            schedule = cron(cronExpression);
        }

        return Tasks.recurring(task.getTaskName(), schedule).execute((instance, ctx) -> task.run());
    }
}
```

**Comment by s-seidel** [comment]
Tested and it works well for us.

**Comment by kagkarlsson** [comment]
Fixed in 11.2

**Comment by donalmurtagh** [comment]
@kagkarlsson tested and confirmed it works, thanks very much
